### DISCUSSION TITLE
Config file is not detected when using async promise with require

### DISCUSSION THREAD

**Discussion by polymorcodeus** [renovatebot/renovate#21292] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.29.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.9

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Currently have 2-3k repos and hitting timeout limits, less than 100 repos are actively using renovate and would like to filter projects based on topic

End result > config.js recognized as valid allowing me to filter

### What you tried so far.

Currently running in gitlab-ci with mirror of https://gitlab.com/renovate-bot/renovate-runner

Tried using the example config [here](https://github.com/renovatebot/renovate/issues/10011#issuecomment-992568583) for running an async function in `config.js`.

Looks like the `got` module is no longer CommonJS supported, however, there's a wrapper `got-cjs` and should be a drop in replacement then with the example. However as soon as I add either `require()` or `import()` renovate stops recognizing the config file.

Based on this open issue - https://github.com/renovatebot/renovate/issues/9890 - I would have expected to need to state in CJS world, but the promise (pun intended) of support for required promises in `config.js` are alluding me.

```
const endpoint = 'https://gitlab.example.io/api/v4/';
const token = process.env.RENOVATE_TOKEN
const topic = "managed-by-renovate";    

const got = require('got-cjs'). // require('doesnt-matter-the-module')

module.exports = async function () {
   ...
}
```

I will mention that if I hard code in the repositories with the `async function ()` defintion inside of `config.js` works as expected, so it's not the async causing the issue, but namely the required(). 

### Relevant debug logs

<details><summary>Logs</summary>

```
Successfully extracted cache
Executing "step_script" stage of the job script
00:06
$ npm install got-cjs -g
added 29 packages in 2s
$ ls
config.js
test.js
$ renovate $RENOVATE_EXTRA_FLAGS
DEBUG: Using RE2 as regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /builds/infra/gitlab/renovate-ci/config.js
DEBUG: No config file found on disk - skipping
DEBUG: Converting GITHUB_COM_TOKEN into a global host rule
DEBUG: Enabling debug logging to renovate-log.ndjson
DEBUG: File config
       "config": {}
DEBUG: CLI config
       "config": {}
DEBUG: Env config
       "config": {
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "***********"}
         ],
```

</details>

**Comment by viceice** [comment]
you can't require renovate modules by default on Gitlab pipelines because the node_modules folder isn't available inside the build directory.

so best is to add a local package.json and add your required packages. then run npm install before renovate (or equivalent package manager)

**Reply by polymorcodeus** [reply]
Not sure I track RE: renovate modules statement

In .gitlab-ci all I'm doing from the config.js project is include the template.
```
include:
  - project: "infra/gitlab/renovate-runner"
    file: "/templates/renovate.gitlab-ci.yml"

variables:
  # Force docker to use TLS
  DOCKER_HOST: "tcp://docker:2376"
  DOCKER_TLS_CERTDIR: "${DOCKER_TLS_CERTDIR}"
  DOCKER_CERT_PATH: "${DOCKER_TLS_CERTDIR}/client"
  # LOG_LEVEL: "debug"
  LOG_LEVEL: "info"
```

Unless you're saying 'npm install got-js' in a `before_script` in my pipeline? I can do a bit more troubleshooting there, because I thought I tried, but not sure I had a package.json with the config.js. 

To be frank, this is my first foray into JS so apologies if I'm asking/stating obvious questions. I do appreciate the patience.

**Reply by viceice** [reply]
you need to install the node modules which are not included in nodejs. you can't reuse the dependencies from renovate.

you can install it via `npm i -g got-cjs@^11` in before script to make the package globally available. 

adapt version of cause.

btw: remove the docker ENV vars, we removed dind support from template to use slim image with install binary source as full compatible replacement.
